js/widgets/musickeyboard.js

```diff
@@ -2638,187 +2638,203 @@ function MusicKeyboard(activity) {
             }
             return ans;
         };
-
-        const newNotes = this._clusterNotes(selectedNotes);
-
-        let prevId = 2;
-        let endOfStackIdx, id;
-
-        for (let noteGrp = 0; noteGrp < newNotes.length; noteGrp++) {
-            const selectedNotesGrp = newNotes[noteGrp];
-            const isLast = noteGrp == newNotes.length - 1;
-            id = newStack.length;
-            let voice = selectedNotes[selectedNotesGrp[0]].voice[0] || DEFAULTVOICE;
-            // Don't use a drum name with set timbre.
-            if (selectedNotes[selectedNotesGrp[0]].noteOctave[0] === "drumnull") {
-                voice = DEFAULTVOICE;
-            }
-            const next = isLast ? null : id + 3 + this.findLen(selectedNotesGrp, selectedNotes);
-
-            newStack.push(
-                [id, "settimbre", 0, 0, [prevId, id + 1, id + 3, id + 2]],
-                [id + 1, ["voicename", { value: voice }], 0, 0, [id]],
-                [id + 2, "hidden", 0, 0, [id, next]]
-            );
-
-            prevId = id + 2;
-            endOfStackIdx = id;
-
-            for (let i = 0; i < selectedNotesGrp.length; i++) {
-                const note = selectedNotes[selectedNotesGrp[i]];
-
-                // Add the Note block and its value
-                const idx = newStack.length;
-                newStack.push([idx, "newnote", 0, 0, [endOfStackIdx, idx + 2, idx + 1, null]]);
-                const n = newStack[idx][4].length;
-                if (i === 0) {
-                    // the action block
-                    newStack[endOfStackIdx][4][n - 2] = idx;
-                } else {
-                    // the previous note block
-                    newStack[endOfStackIdx][4][n - 1] = idx;
+        const actionGroupInterval = 50;
+        var actionGroups = parseInt((selectedNotes.length)/actionGroupInterval)+1;
+        
+        for (let actionGroup = 0;actionGroup<actionGroups;actionGroup++){
+            let currentSelectedNotes = selectedNotes.slice(actionGroup*actionGroupInterval,(actionGroup+1)*actionGroupInterval);
+        
+            const newNotes = this._clusterNotes(currentSelectedNotes);
+            const newStack = [
+                [0, ["action", { collapsed: false }], 100, 100, [null, 1, 2, null]],
+                [1, ["text", { value: _("action")+""+actionGroup }], 0, 0, [0]],
```

Comment:
Why the +""+ ???

Reply:
![Screenshot from 2023-12-29 00-44-01](https://github.com/sugarlabs/musicblocks/assets/118426425/b08c8836-fc5a-4673-9f82-999d4ebca078)
you mentioned it in one of the pr reviews, so i just added it